cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/documentstore/DatabaseImplExceptionsTest.java

```diff
@@ -0,0 +1,196 @@
+package com.cloudant.sync.internal.documentstore;
+
+import com.cloudant.sync.documentstore.DocumentBodyFactory;
+import com.cloudant.sync.documentstore.DocumentException;
+import com.cloudant.sync.documentstore.DocumentNotFoundException;
+import com.cloudant.sync.documentstore.DocumentRevision;
+import com.cloudant.sync.documentstore.DocumentStoreException;
+import com.cloudant.sync.internal.sqlite.SQLCallable;
+import com.cloudant.sync.internal.sqlite.SQLDatabase;
+import com.cloudant.sync.util.TestUtils;
+
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+/**
+ * Created by tomblench on 23/11/2016.
+ */
+
+
+// tests to check that the correct exceptions get thrown for each public API call eg when the
+// underlying DB has had critical tables dropped
+
+public class DatabaseImplExceptionsTest extends BasicDatastoreTestBase {
+
+    private void dropRevs() {
+        this.datastore.runOnDbQueue(new SQLCallable<Void>() {
+            public Void call(SQLDatabase db) {
+                try {
+                    db.execSQL("DROP TABLE revs;");
+                } catch (SQLException sqe) {
+                    ;
+                }
+                return null;
+            }
+        });
+    }
+
+    // getDocument on non-existent document should throw DocumentNotFoundException
+    @Test(expected = DocumentNotFoundException.class)
+    public void getDocumentShouldThrowDocumentNotFoundException() throws Exception {
+        this.datastore.getDocument("nosuchdocument");
+    }
+
+    // getDocument after we remove the underlying SQL database should throw DocumentStoreException
+    @Test(expected = DocumentStoreException.class)
+    public void getDocumentShouldThrowDocumentStoreException() throws Exception {
+        // remove the underlying database, triggering a SQL Exception
+        dropRevs();
+        this.datastore.getDocument("nosuchdocument");
+    }
+
+    // getDocument on non-existent revid should throw DocumentNotFoundException
+    @Test(expected = DocumentNotFoundException.class)
+    public void getDocumentWithRevisionShouldThrowDocumentNotFoundException() throws Exception {
+        DocumentRevision dr = new DocumentRevision("doc1");
+        dr.setBody(DocumentBodyFactory.create("{\"hello\":\"world\"}".getBytes()));
+        this.datastore.createDocumentFromRevision(dr);
+        this.datastore.getDocument("nosuchdocument", "nosuchrevid");
```

Comment:
Shouldn't we use the valid doc id `doc1` here instead of `nosuchdocument` or the test might be passing because of an invalid doc id rather than an invalid rev id.